### PR TITLE
add exit code

### PR DESCRIPTION
**Pull request by ssabo**
**What this PR does / why we need it**:

Adds a new metric with the exit code from LastTerminantedStatus. 

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*

A new metric will be added with the same cardinality as `kube_pod_container_status_last_terminated_reason` but with a value of the exit code

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
https://github.com/kubernetes/kube-state-metrics/issues/1481

### CODE REVIEW

**Comment by fpetkovski** [comment]
Thanks for the contribution. As discussed in one of the previous issues, I think this should be a new metric where the exit code is a value. Otherwise the cardinality impact will be very high.

If you need to filter by exit codes, you could always add `== <exit_code>` at the end of a PromQL expression.

**Comment by ssabo** [comment]
Ah I hadn't realized that PromQL supported that grammar. That makes perfect sense to me, I'll make those updates.

File: internal/store/pod.go [hunk 1]
```diff
@@ -334,6 +335,31 @@ func createPodContainerStatusLastTerminatedReasonFamilyGenerator() generator.Fam
...
+		"kube_pod_container_status_last_terminated_exitcode",
+		"Describes the exit code for the last container in terminated state.",
+		metric.Gauge,
+		"",
+		wrapPodFunc(func(p *v1.Pod) *metric.Family {
+			ms := make([]*metric.Metric, 0, len(p.Status.ContainerStatuses))
+			for _, cs := range p.Status.ContainerStatuses {
+				if cs.LastTerminationState.Terminated != nil {
+					ms = append(ms, &metric.Metric{
+						LabelKeys:   []string{"container", "reason"},
```
Comment (ssabo) [inline internal/store/pod.go:349]: Should I keep reason attached to the exit code as a tag? or remove it entirely?

Comment (fpetkovski) [reply internal/store/pod.go:349]: I think we should remove it because it's already present in a different metric

**Comment by fpetkovski** [comment]
This looks good, but the CI needs to be fixed before we can merge it.

**Comment by ssabo** [comment]
@fpetkovski i'll probably get some some time to look at fixing the CI tests next week. In the meantime, what's the best way to run those locally? I don't think I have the capability to do `/retest` in this project yet and I was running into missing CLI tools when I had tried to invoke them manually. Any way to run them containerized? or just have to install those dependencies?

**Comment by fpetkovski** [comment]
You can check directly what the CI commands are. It seems that [`make test-benchmark-compare`](https://github.com/kubernetes/kube-state-metrics/runs/6873557351?check_suite_focus=true#step:5:2) and [`make doccheck`](https://github.com/kubernetes/kube-state-metrics/runs/6873557547?check_suite_focus=true#step:5:2) are failing.

**Comment by ssabo** [comment]
@fpetkovski sorry for the long delay. Life got busy unexpectedly. Tests should be all fixed up now

**Review by fpetkovski — COMMENTED** [review]
Thanks

/lgtm

**Pull request merged**